File: sdk/samples/WCFSamples/TechnologySamples/Basic/WorkflowServices/WorkflowFirstAndSecurity/VB/Client/SequentialCalculatorClient.xoml.vb
' Copyright (c) Microsoft Corporation. All rights reserved.

Imports System
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Collections
Imports System.Workflow.ComponentModel.Compiler
Imports System.Workflow.ComponentModel.Serialization
Imports System.Workflow.ComponentModel
Imports System.Workflow.ComponentModel.Design
Imports System.Workflow.Runtime
Imports System.Workflow.Activities
Imports System.Workflow.Activities.Rules

Namespace Microsoft.WorkflowServices.Samples
    Partial Public Class SequentialCalculatorClient
        Inherits SequentialWorkflowActivity
        Private random As New Random()
        Public outputValue As Integer = Nothing
        Public returnValue As Integer = Nothing

        Private Sub OnBeforeSend(ByVal sender As Object, ByVal e As SendActivityEventArgs)
            outputValue = random.[Next](1, 100)
            Console.WriteLine(e.SendActivity.Name + "(" + outputValue.ToString() + ")")
        End Sub

        Private Sub OnAfterResponse(ByVal sender As Object, ByVal e As SendActivityEventArgs)
            Console.WriteLine("[" + returnValue.ToString() + "]")
        End Sub
    End Class
End Namespace

File: sdk/samples/WCFSamples/TechnologySamples/Basic/WorkflowServices/WorkflowFirstAndSecurity/CS/Client/SequentialCalculatorClient.xoml.cs
//  Copyright (c) Microsoft Corporation. All rights reserved.

using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;

namespace Microsoft.WorkflowServices.Samples
{
	public partial class SequentialCalculatorClient : SequentialWorkflowActivity
	{
        Random random = new Random();
        public int outputValue = default(int);
        public int returnValue = default(int);

        private void OnBeforeSend(object sender, SendActivityEventArgs e)
        {
            outputValue = random.Next(1, 100);
            Console.WriteLine(e.SendActivity.Name + "(" + outputValue.ToString() + ")");
        }

        private void OnAfterResponse(object sender, SendActivityEventArgs e)
        {
            Console.WriteLine("[" + returnValue.ToString() + "]");
        }
    }
}
